.changeset/tender-falcons-add.md

```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/cli': patch
+---
+
+Fixed the dynamic import of vite.
```

Comment:
Do you know if there's any particular version of `vite` where the behavior changes? Thinking it would be good if we could highlight that in the changeset. What version are you testing with?

Reply:
Testing with these versions
```json
"@vitejs/plugin-react": "^4.0.4",
"vite": "^4.4.9",
"vite-plugin-html": "^3.2.0",
"vite-plugin-node-polyfills": "^0.21.0"
```

Reply:
Which seems to match the backstage cli vite versions: https://github.com/backstage/backstage/blob/95ee6d6a2447a301cfc6c9a2bcfa84f9171e9bf9/packages/cli/package.json#L188

Reply:
Which version is it actually resolved to though in `yarn.lock`?

Reply:
our own backstage instance:
```ts
└─ app@workspace:packages/app
   └─ vite@npm:4.5.3 [f87a9] (via npm:^4.4.9 [f87a9])
```

while in my fork on this branch:
```ts
├─ @backstage/cli@workspace:packages/cli [05476]
│  └─ vite@npm:4.5.3 [23bd4] (via npm:^4.4.9 [23bd4])
│
├─ @backstage/cli@workspace:packages/cli [07278]
│  └─ vite@npm:4.5.3 [23bd4] (via npm:^4.4.9 [23bd4])
│
├─ @backstage/cli@workspace:packages/cli [11fe3]
│  └─ vite@npm:4.5.3 [23bd4] (via npm:^4.4.9 [23bd4])
│
├─ @backstage/cli@workspace:packages/cli [31634]
│  └─ vite@npm:4.5.3 [23bd4] (via npm:^4.4.9 [23bd4])
│
├─ @backstage/cli@workspace:packages/cli [39bca]
│  └─ vite@npm:4.5.3 [23bd4] (via npm:^4.4.9 [23bd4])
│
├─ @backstage/cli@workspace:packages/cli [70735]
│  └─ vite@npm:4.5.3 [23bd4] (via npm:^4.4.9 [23bd4])
│
├─ @backstage/cli@workspace:packages/cli [aad9f]
│  └─ vite@npm:4.5.3 [23bd4] (via npm:^4.4.9 [23bd4])
│
├─ @backstage/cli@workspace:packages/cli [b9c15]
│  └─ vite@npm:4.5.3 [23bd4] (via npm:^4.4.9 [23bd4])
│
├─ @backstage/cli@workspace:packages/cli [c856b]
│  └─ vite@npm:4.5.3 [23bd4] (via npm:^4.4.9 [23bd4])
│
├─ example-app-next@workspace:packages/app-next
│  └─ vite@npm:4.5.3 [d47fc] (via npm:^4.4.9 [d47fc])
│
└─ example-app@workspace:packages/app
   └─ vite@npm:4.5.3 [d47fc] (via npm:^4.4.9 [d47fc])
```

Reply:
I wonder if we should actually bite the bullet and bump to `vite@v5`? Version 6 is on it's way soon, and from what I can tell I don't think that there's any breaking changes that we would run into.